### PR TITLE
fix: openurl_if_required enhancement

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

`openurl_if_required` added

@sai-harsha-vardhan 

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/Utilities/PaymentHelpers.res [hunk 1]
```diff
@@ -357,6 +356,7 @@ let rec intentCall = (
     let url = urlSearch(confirmParam.return_url)
     url.searchParams.set("payment_intent_client_secret", clientSecret)
     url.searchParams.set("status", "failed")
+    handlePostMessage([("confirmParamsPostMessage", confirmParam->Identity.anyTypeToJson)])
```
Comment (prafulkoppalkar) [inline src/Utilities/PaymentHelpers.res:359]:
```suggestion
    handlePostMessage([("confirmParams", confirmParam->Identity.anyTypeToJson)])
```

File: src/orca-loader/Elements.res [hunk 2]
```diff
@@ -623,17 +628,21 @@ let make = (
...
+                    resolve(JSON.Encode.null)
+                  } else {
+                    resolve(json)
+                  }
                 })
-                ->catch(_ => {
+                ->catch(err => {
                   Window.Location.replace(url)
-                  resolve()
+                  resolve(err->Identity.anyTypeToJson)
```
Comment (prafulkoppalkar) [inline src/orca-loader/Elements.res:645]: we need to add the same check here too ..

**Review by prafulkoppalkar — CHANGES_REQUESTED** [review]
requested changes

File: src/orca-loader/Elements.res [hunk 3]
```diff
@@ -28,6 +28,7 @@ let make = (
     let savedPaymentElement = Dict.make()
     let localOptions = options->JSON.Decode.object->Option.getOr(Dict.make())
     let endpoint = ApiEndpoint.getApiEndPoint(~publishableKey, ())
+    let redirect = ref("if_required")
```
Comment (ArushKapoorJuspay) [inline src/orca-loader/Elements.res:31]: There is no need to have redirect as a ref

File: src/Types/ConfirmType.res [hunk 4]
```diff
@@ -1,6 +1,7 @@
 type confirmParams = {
   return_url: string,
   publishableKey: string,
+  redirect?: string,
```
Comment (ArushKapoorJuspay) [inline src/Types/ConfirmType.res:4]: There is no need to have redirect as an optional param

Comment (PritishBudhiraja) [reply src/Types/ConfirmType.res:4]: It's required else unnecessary I have to pass if_required everywhere

File: src/orca-loader/Elements.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/orca-loader/Elements.res:31]: It's required to set the value in that context and use this.

File: src/orca-loader/Elements.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/orca-loader/Elements.res:652]: updated

**Review by prafulkoppalkar — APPROVED** [review]

File: src/Types/ConfirmType.res (hunk 4 above)
Comment (PritishBudhiraja) [reply src/Types/ConfirmType.res:4]: It's not required for wallets

**Review by ArushKapoorJuspay — APPROVED** [review]

**Pull request merged**